src/compiler/parser.ts

```diff
@@ -3225,6 +3228,16 @@ namespace ts {
             }
         }
 
+        function parseAssertsTypePredicate(): TypeNode {
+            const node = <TypePredicateNode>createNode(SyntaxKind.TypePredicate);
+            node.assertsModifier = parseExpectedToken(SyntaxKind.AssertsKeyword);

```

Comment:
adding this property here and not assigning it in `parseTypeOrTypePredicate` where regular `TypePredicate` nodes are constructed, create yet another hidden class that hinders optimization at runtime.
Either assign it last in this function or (even better) assign it in both functions in the same order

Reply:
Agreed